internal/telemetry/client.go

```diff
@@ -119,8 +119,10 @@ type client struct {
 	// http.DefaultTransport and a 5 second timeout will be used.
 	Client *http.Client
 
+	// productSync syncs profiling and tracer start/stop
+	productSync sync.Mutex
```

Comment:
@ajgajg1134 during our pair programming session I know we just used `mu` (the mutex that was already there) to sync `ProductStop` and `ProductStart`, but I ran into some troubles with race conditions in the unit tests that directly called the now un-exported `start`. What do you think of just having a separate lock like this for syncing product start/stop?